lib/countly.js

```diff
@@ -272,1545 +247,2505 @@
+                }
+
+                return self.deserialize(data);
+
+            }
+
+            // Null specified, remove store
+            if (value === null) {
+                if (lsSupport) { // Native support
+                    localStorage.removeItem(key);
+                }
+                else if (!storageOnly) { // Use cookie
+                    createCookie(key, "", -1);
+                }
+            }
+
+            /**
+             *  Creates new cookie or removes cookie with negative expiration
+             *  @param {String} cookieKey - The key or identifier for the store
+             *  @param {String} cookieVal - Contents of the store
+             *  @param {Number} exp - Expiration in days
+             */
+            function createCookie(cookieKey, cookieVal, exp) {
+                var date = new Date();
+                date.setTime(date.getTime() + (exp * 24 * 60 * 60 * 1000));
+                var expires = "; expires=" + date.toGMTString();
+                document.cookie = cookieKey + "=" + cookieVal + expires + "; path=/";
+            }
+
+            /**
+            * Returns contents of cookie
+            * @param {String} cookieKey - The key or identifier for the store
+            * @returns {Varies} stored value
+            */
+            function readCookie(cookieKey) {
+                var nameEQ = cookieKey + "=";
+                var ca = document.cookie.split(";");
+                for (var i = 0, max = ca.length; i < max; i++) {
+                    var c = ca[i];
+                    while (c.charAt(0) === " ") {
+                        c = c.substring(1, c.length);
+                    }
+                    if (c.indexOf(nameEQ) === 0) {
+                        return c.substring(nameEQ.length, c.length);
+                    }
+                }
+                return null;
+            }
+        }
+
+        /**
+         *  Migrate from old storage to new app_key prefixed storage
+         */
+        function migrate() {
+            if (store(self.namespace + "cly_id", undefined, false, true)) {
+                //old data exists, we should migrate it
+                store("cly_id", store(self.namespace + "cly_id", undefined, false, true));
+                store("cly_event", store(self.namespace + "cly_event", undefined, false, true));
+                store("cly_session", store(self.namespace + "cly_session", undefined, false, true));
+
+                //filter out requests with correct app_key
+                var requests = store(self.namespace + "cly_queue", undefined, false, true);
+                if (Array.isArray(requests)) {
+                    requests = requests.filter(function(req) {
+                        return req.app_key === self.app_key;
+                    });
+                    store("cly_queue", requests);
+                }
+                if (store(self.namespace + "cly_cmp_id", undefined, false, true)) {
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/manual/ar2rsawseen/countly-sdk-web/pullRequest?prid=6356797)